src/js/utils/strings.js

```diff
@@ -99,6 +99,17 @@ export function seconds(str, frameRate) {
     return sec;
 }
 
+// Convert an offset string to a number; supports conversion of percentage offsets
+export function offsetToSeconds(offset, duration, frameRate) {
+    if (_.isString(offset) && offset.slice(-1) === '%') {
+        const percent = parseFloat(offset);
+        if (duration && !isNaN(percent)) {
+            return duration * percent / 100;
+        }
+        return 0;
```

Comment:
We should not return 0 if duration is unknown. The result is not an offset of `0` seconds. We should return `null` and then so code using this util knows to wait for a valid duration.